R/antolini.R

```diff
@@ -0,0 +1,45 @@
+cindex = function(pred, meth = c("A", "H"), tiex = 0.5) {
+  times = pred$truth[, 1]
+  outcome = pred$truth[, 2]
+
+  if (meth == "A") {
+    scores = diag(distr6:::C_Vec_WeightedDiscreteCdf(times, sort(unique(times)),
```

Comment:
So instead of `sort(unique(times)` (which are only the times for the test set) I would do `as.numeric(colnames(surv)`, where `surv = pred$data$distr`, because that's the ones you want there (and these usually are the sort, unique times from the train set, ie the ones the trained algorithm has used in the `pred$data$distr` survival matrix). You would see this if you checked the `dim` of the returned matrix from the `distr6` function => **it's not a square matrix**, but the `diag` works (which is bad in this case)

Reply:
I committed a fixed version for this